### PR TITLE
Custom Connection Timeout in DriverRemoteConnection

### PR DESCRIPTION
**Pull request by L0Lmaker**
*Issue #, if available:*
https://bitquill.atlassian.net/jira/software/c/projects/AN/boards/36?modal=detail&selectedIssue=AN-986

*Description of changes:*
Custom Connection Timeout in DriverRemoteConnection

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gremlin-go/driver/client.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ type ClientSettings struct {
 	// Maximum number of concurrent connections. Default: number of runtime processors
 	MaximumConcurrentConnections int
 	Session                      string
+	ConnectionTimeout            time.Duration
```
Comment (simonz-bq) [inline gremlin-go/driver/client.go]: Can you please move ConnectionTimeout in the various properties and method parameters to be under WriteDeadline for better oganization?

Other than that, LGTM

**Review by lyndonbauto — DISMISSED** [review]

File: gremlin-go/driver/connectionPool_test.go [hunk 2]
```diff
@@ -35,8 +35,7 @@ var logger = newLogHandler(&defaultLogger{}, Info, language.English)
 func getPoolForTesting() *loadBalancingPool {
 	return &loadBalancingPool{
 		url:                    "",
-		authInfo:               nil,
-		tlsConfig:              nil,
+		connSettings:           newDefaultConnectionSettings(),
```
Comment (L0Lmaker) [inline gremlin-go/driver/connectionPool_test.go:38]: Do we need to maintain nil for tests? I can update this to make `authInfo` and `tlsConfig` in `connSettings` nil if needed but the tests seem to pass.

**Comment by vkagamlyk** [comment]
need to resolve conflicts

**Review by vkagamlyk — DISMISSED** [review]
LGTM

File: gremlin-go/driver/client.go [hunk 3]
```diff
@@ -38,13 +38,22 @@ type ClientSettings struct {
...
 	WriteDeadline     time.Duration
+	ConnectionTimeout time.Duration
 	// Minimum amount of concurrent active traversals on a connection to trigger creation of a new connection
 	NewConnectionThreshold int
 	// Maximum number of concurrent connections. Default: number of runtime processors
 	MaximumConcurrentConnections int
 	Session                      string
 }
 
+type connectionSettings struct {
```
Comment (simonz-bq) [inline gremlin-go/driver/client.go]: Would recommend that this lives in `connection.go`

**Review by simonz-bq — DISMISSED** [review]

**Review by vkagamlyk — APPROVED** [review]

**Review by lyndonbauto — APPROVED** [review]
